pos_order_mgmt/static/src/js/widgets.js

```diff
@@ -0,0 +1,406 @@
+                var line = paymentLine;
+                // In case of local data
+                if (line.length === 3) {
+                    line = line[2];
+                }
+                _.each(self.pos.cashregisters, function (cashregister) {
+                    if (cashregister.id === line.statement_id) {
+                        if (line.amount > 0) {
+                            // If it is not change
+                            order.add_paymentline(cashregister);
+                            order.selected_paymentline.set_amount(line.amount);
+                        }
+                    }
+                });
+            });
+            return order;
+        },
+        _prepare_orderlines_from_order_data: function(order, order_data, orderLines) {
+            var self = this;
+            _.each(orderLines, function(line) {
+                // In case of local data
+                if (line.length === 3) {
+                    line = line[2];
+                }
+                var product = self.pos.db.get_product_by_id(line.product_id);
+                // Check if product are available in pos
+                if (_.isUndefined(product)) {
+                    self.unknown_products.push(String(line.product_id));
+                } else {
+                    // Create a new order line
+                    order.add_product(product, {
+                        price: line.price_unit,
+                        quantity: order_data.return ? line.qty * -1 : line.qty,
+                        discount: line.discount,
+                        merge: false,
+                    });
+                }
+            });
+        },
+
+        load_order: function (order_id, action) {
+            this.unknown_products = [];
+            var self = this;
+            return this._rpc({
+                model: 'pos.order',
+                method: 'load_done_order_for_pos',
+                args: [order_id],
+            }).then(function (order_data) {
+                self.gui.show_screen('orderlist');
+                var correct_order_print = true;
+                if (action === 'return') {
+                    order_data.return = true;
+                }
+                var order = self._prepare_order_from_order_data(order_data);
+                // Forbid POS Order loading if some products are unknown
+                if (self.unknown_products.length > 0) {
+                    self.gui.show_popup('error-traceback', {
+                        'title': _t('Unknown Products'),
+                        'body': _t('Unable to load some order lines because the ' +
+                            'products are not available in the POS cache.\n\n' +
+                            'Please check that lines :\n\n  * ') + self.unknown_products.join("; \n  *"),
+                    });
+                    correct_order_print = false;
+                }
+                if (correct_order_print && action === 'print') {
```

Comment:
Hi,

I'm using this code for a project and wanted to ask if it is possible to change this part to be a little easier to extend. For example by putting it in some kind of dispatch method or like this:

```suggestion
                if(correct_order_print){
                    self["action_" + action](order);
                }
```

**Pros**: Easier to extend (You only need to register the event and write the method)
**Cons**: If you need different checks before executing the action or you want to use actions not starting with action this suggestion won't work. (But the dispatch method still would.)

Reply:
Also if there is more work to do with this module I would be glad to help :)

Reply:
Hi @Kerrim- , No problem about accepting a PR from my side :+1: I only use the module for reprinting tickets so I did not dig a lot into returns